### PR TITLE
[oap-native-sql]Add ColumnarBatch Combination on Shuffle Read Side

### PR DESCRIPTION
**Pull request by rongma1997**
## What changes were proposed in this pull request?

Add a new operator CoalesceBatch to support ColumnarBatch combination on shuffle read side.

## How was this patch tested?

Existing unit tests. Manually test and pass all tpch queries.

### CODE REVIEW

**Comment by zhixingheyi-tian** [comment]
@rongma1997 
Would you please add the module name  in the PR title?

Thanks

File: oap-native-sql/core/src/main/scala/com/intel/sparkColumnarPlugin/execution/CoalesceBatchesExec.scala [hunk 1]
```diff
@@ -1,23 +1,152 @@
...
+
+    val recordsPerBatch = conf.arrowMaxRecordsPerBatch
+    val numInputRows = longMetric("numInputRows")
+    val numInputBatches = longMetric("numInputBatches")
+    val numOutputBatches = longMetric("numOutputBatches")
+    val concatTime = longMetric("concatTime")
+    val collectTime = longMetric("collectTime")
+    val avgCoalescedNumRows = longMetric("avgCoalescedNumRows")
+
+    child.executeColumnar().mapPartitions { iter =>
```
Comment (zhouyuan) [inline oap-native-sql/core/src/main/scala/com/intel/sparkColumnarPlugin/execution/CoalesceBatchesExec.scala:65]: I tried to increase this value to 20k, but from the statics of the operator, it's able to get 5k rows/batch. Is this the expected behavior?

Comment (rongma1997) [reply oap-native-sql/core/src/main/scala/com/intel/sparkColumnarPlugin/execution/CoalesceBatchesExec.scala:65]: which value?

Comment (zhouyuan) [reply oap-native-sql/core/src/main/scala/com/intel/sparkColumnarPlugin/execution/CoalesceBatchesExec.scala:65]: (clicked on the wrong line) 
I mean for this line L57
`val recordsPerBatch = conf.arrowMaxRecordsPerBatch`

Comment (rongma1997) [reply oap-native-sql/core/src/main/scala/com/intel/sparkColumnarPlugin/execution/CoalesceBatchesExec.scala:65]: Yes. For each reducer task, the actual combined size can be smaller than the configuration if there are no more data to get.

**Pull request merged**